django/db/models/query.py

```diff
@@ -322,6 +322,12 @@ def aggregate(self, *args, **kwargs):
         """
         if self.query.distinct_fields:
             raise NotImplementedError("aggregate() + distinct(fields) not implemented.")
+        invalid_args = [str(arg) for arg in args + tuple(kwargs.values()) if not self._is_expression(arg)]
```

Comment:
I was thinking all of this could be in a method like:
```python
def validate_values_are_expressions(values, method_name):
    invalid_args = sorted(str(arg) for arg in values if not hasattr(arg, 'resolve_expression')])
    if invalid_args:
        raise TypeError(
            'Queryset.%s() received non-expression(s): %s.' % (
                method_name,
                ', '.join(invalid_args),
            )
        )
```